quill-core/src/main/scala/io/getquill/norm/ExpandReturning.scala

```diff
@@ -21,7 +16,7 @@ object ExpandReturning {
     val initialExpand = ExpandReturning.apply(returning)(idiom, naming)
 
     idiom.idiomReturningCapability match {
-      case ReturningClauseSupported =>
+      case ReturningClauseSupported | OutputClauseSupported =>
```

Comment:
This is good, but down below, change this:
```scala
def apply(returning: ReturningAction)(idiom: Idiom, naming: NamingStrategy): List[(Ast, Statement)] = {

    // Ident("j"), Tuple(List(Property(Ident("j"), "name"), BinaryOperation(Property(Ident("j"), "age"), +, Constant(1))))
    // => Tuple(List(ExternalIdent("name"), BinaryOperation(ExternalIdent("age"), +, Constant(1))))
    val dePropertized =
      Transform(properties) {
        case `alias` => ExternalIdent(alias.name)
      }
```
Into this:
```scala
  def apply(returning: ReturningAction, renameAlias: Option[String] = None)(idiom: Idiom, naming: NamingStrategy): List[(Ast, Statement)] = {
    val ReturningAction(_, alias, properties) = returning

    // Ident("j"), Tuple(List(Property(Ident("j"), "name"), BinaryOperation(Property(Ident("j"), "age"), +, Constant(1))))
    // => Tuple(List(ExternalIdent("name"), BinaryOperation(ExternalIdent("age"), +, Constant(1))))
    val dePropertized = renameAlias match {
      case Some(newName) =>
        Transform(properties) {
          case `alias` => ExternalIdent(newName)
        }
      case None =>
        Transform(properties) {
          case `alias` => ExternalIdent(alias.name)
        }
    }
```
Alternatively, instead of a `Transform`, `BetaReduction` might be a more Quill-holistic way to do it but I think you should double check if it works right.
```scala
  def apply(returning: ReturningAction, renameAlias: Option[String] = None)(idiom: Idiom, naming: NamingStrategy): List[(Ast, Statement)] = {
    val ReturningAction(_, alias, properties) = returning

    // Ident("j"), Tuple(List(Property(Ident("j"), "name"), BinaryOperation(Property(Ident("j"), "age"), +, Constant(1))))
    // => Tuple(List(ExternalIdent("name"), BinaryOperation(ExternalIdent("age"), +, Constant(1))))
    val dePropertized = renameAlias match {
      case Some(newName) =>
        BetaReduction(properties, alias -> ExternalIdent(newName))
      case None =>
        BetaReduction(properties, alias -> ExternalIdent(alias.name))
    }
```

Reply:
One other thing, because `initialExpand` is not being used by every clause,  I think it would be much more clear if we move it into the respective clause:
```scala
  def applyMap(returning: ReturningAction)(f: (Ast, Statement) => String)(idiom: Idiom, naming: NamingStrategy) = {
    idiom.idiomReturningCapability match {
      case ReturningClauseSupported =>
        ReturnAction.ReturnRecord
      case ReturningMultipleFieldSupported =>
        val initialExpand = ExpandReturning.apply(returning)(idiom, naming)
        ReturnColumns(initialExpand.map { case (ast, statement) => f(ast, statement) })
      case ReturningSingleFieldSupported =>
        val initialExpand = ExpandReturning.apply(returning)(idiom, naming)
        if (initialExpand.length == 1)
          ReturnColumns(initialExpand.map { case (ast, statement) => f(ast, statement) })
        else
          throw new IllegalArgumentException(s"Only one RETURNING column is allowed in the ${idiom} dialect but ${initialExpand.length} were specified.")
      case ReturningNotSupported =>
        throw new IllegalArgumentException(s"RETURNING columns are not allowed in the ${idiom} dialect.")
    }
  }
```